packages/app-builder-lib/templates/nsis/include/allowOnlyOneInstallerInstance.nsh

```diff
@@ -62,16 +62,34 @@
       # to ensure that files are not "in-use"
       Sleep 300
 
-      ${nsProcess::FindProcess} "${APP_EXECUTABLE_FILENAME}" $R0
-      ${if} $R0 == 0
-        # wait to give a chance to exit gracefully
-        Sleep 1000
-        nsExec::Exec `taskkill /f /im "${APP_EXECUTABLE_FILENAME}" /fi "PID ne $pid"` $R0
-        ${If} $R0 != 0
-          DetailPrint `Waiting for "${PRODUCT_NAME}" to close (taskkill exit code $R0).`
-          Sleep 2000
+      # Retry counter
+      StrCpy $R1 0
+
+      loop:
+        IntOp $R1 $R1 + 1
+
+        ${nsProcess::FindProcess} "${APP_EXECUTABLE_FILENAME}" $R0
+        ${if} $R0 == 0
+          # wait to give a chance to exit gracefully
+          Sleep 1000
```

Comment:
Was this 1s an arbitrary choice? I'm worried it's not enough time for low-performance hardware.

Is there an event we can listen for? Or at least have a retry-loop for several seconds to check if the process exists?

Reply:
The logic in question was there before my changes, and the only difference between original code and new one is that it will ask for user confirmation and attempt to retry indefinitely (each time with a confirmation) until success. So the scope of this PR in short is to not let installation succeed while the executable is still locked.

As a follow up PR, I can try using `nsProcess::CloseProcess`, which closes all process windows waits for up to 3 seconds for the process to actually exit until it kills it. I'm not exactly sure if this is what we want though, since it is not unlikely that the electron process might ask user if they want to save some data before closing and 3 seconds is not enough for this to finish.